### PR TITLE
Adds paths-ignore to the Github pull request workflow

### PR DESCRIPTION
**Pull request by derobins**

### CODE REVIEW

File: .github/workflows/pr-check.yml [hunk 1]
```diff
@@ -4,7 +4,14 @@ name: hdf5 dev CI
...
   pull_request:
     branches: [ develop, hdf5_1_12, hdf5_1_10, hdf5_1_8 ]
-
+    paths-ignore:
+    - '.github/**'
+    - 'doc/**'
+    - 'release_docs/**'
+    - 'ACKNOWLEDGEMENTS'
+    - 'COPYING**'
+    - '**.md'
```
Comment (derobins) [inline .github/workflows/pr-check.yml:13]: Nothing new here - these were copied from main.yml

**Review by byrnHDF — APPROVED** [review]

File: .github/workflows/pr-check.yml (hunk 1 above)
Comment (hyoklee) [reply .github/workflows/pr-check.yml:13]: Will it skip spelling tests on *.md files?

Comment (derobins) [reply .github/workflows/pr-check.yml:13]: That's a good question. Let me check to see if we can always run the spelling checks but not the code checks.

Comment (byrnHDF) [reply .github/workflows/pr-check.yml:13]: spelling, format and code checks are all separate actions. Code and format are split into two actions - one for push and one for pull-requests

**Review by hyoklee — APPROVED** [review]

**Comment by hyoklee** [comment]
@derobins or @byrnHDF , can you change the name of this Action on the first line? E.g.,

`name: PR hdf5 dev CI`

It's confusing to have the same name as `main.yml` Action.

**Comment by derobins** [comment]
> @derobins or @byrnHDF , can you change the name of this Action on the first line? E.g.,
> 
> `name: PR hdf5 dev CI`
> 
> It's confusing to have the same name as `main.yml` Action.

I'll take care of this in a separate PR.

**Review by byrnHDF — APPROVED** [review]

**Review by lrknox — APPROVED** [review]

**Pull request merged**